nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -428,14 +436,49 @@ public static void checkTypeParameterNullnessForAssignability(
       AssignmentTree assignmentTree = (AssignmentTree) tree;
       rhsTree = assignmentTree.getExpression();
     }
+
+    if (rhsTree instanceof MethodInvocationTree) {
+      MethodInvocationTree methodInvocationTree = (MethodInvocationTree) rhsTree;
+      Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(methodInvocationTree);
+      // update inferredTypes cache for assignments
+      // generic method call with no explicit generic arguments
+      if (methodSymbol.type instanceof Type.ForAll
+          && methodInvocationTree.getTypeArguments().isEmpty()) {
+        Type returnType = methodSymbol.getReturnType();
+        Map<TypeVariable, Type> genericNullness =
+            returnType.accept(new InferTypeVisitor(config), lhsType);
+        if (genericNullness != null) {
+          inferredTypes.put(methodInvocationTree, genericNullness);
+        }
+      }
+    }
     // rhsTree can be null for a VariableTree.  Also, we don't need to do a check
     // if rhsTree is the null literal
     if (rhsTree == null || rhsTree.getKind().equals(Tree.Kind.NULL_LITERAL)) {
       return;
     }
     Type rhsType = getTreeType(rhsTree, config);
 
-    if (lhsType != null && rhsType != null) {
+    if (rhsTree instanceof MethodInvocationTree) {
+      // recreate rhsType using inferred types
+      MethodInvocationTree methodInvocationTree = (MethodInvocationTree) rhsTree;
+      Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(methodInvocationTree);
+      if (inferredTypes.containsKey(methodInvocationTree)) {
+        Map<TypeVariable, Type> genericNullness = inferredTypes.get(methodInvocationTree);
+        List<Type> keyTypeList =
+            genericNullness.keySet().stream()
+                .map(typeVar -> (Type) typeVar)
+                .collect(Collectors.toList());
+        com.sun.tools.javac.util.List<Type> from = com.sun.tools.javac.util.List.from(keyTypeList);
+        com.sun.tools.javac.util.List<Type> to =
+            com.sun.tools.javac.util.List.from(genericNullness.values());
```

Comment:
From this code, it's unclear that the `from` and `to` lists will end up in the appropriate order, where the right keys are lined up with the right values.  (It might work with `LinkedHashMap`s but the code doesn't use those or enforce that right now.)  Instead, I suggest looping over the _entries_ of `genericNullness`, and building the lists one key-value pair at a time.  You can build up the right list type using a `ListBuffer` and calling `append` on it, like here: https://github.com/uber/NullAway/blob/15c817aa672e712416bc0a4ca5ce46162845fc2e/nullaway/src/main/java/com/uber/nullaway/generics/TypeSubstitutionUtils.java#L181-L181